src/components/DuplicateCourseModal.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useQuery } from 'react-query';
+
+import { Button, Checkbox, Modal, Select } from '@folio/stripes/components';
+import { useOkapiKy } from '@folio/stripes/core';
+
+
+import manifest from '../util/manifest';
+
+const DuplicateCourseModal = ({ data, history, onClose }) => {
+  const intl = useIntl();
+  const ky = useOkapiKy();
+
+  const [term, setTerm] = useState();
+  const { data: terms } = useQuery(
+    ['ui-courses', 'terms'],
+    () => ky(manifest.terms.path).json(),
+    {
+      placeholderData: [],
+      onSuccess: json => setTerm(json[0]?.value),
+      select: json => ((json?.terms ?? [])
+        .map(t => ({ value: t.id, label: t.name }))
+        .sort((a, b) => a.label.localeCompare(b.label))
+      )
+    }
+  );
+
+  const [duplicateCrosslistedCourses, setDuplicateCrosslistedCourses] = useState(false);
+  useQuery(
+    ['ui-courses', 'duplicateCrosslistedCourses'],
+    () => ky(manifest.displaySettings.path).json(),
+    {
+      onSuccess: json => setDuplicateCrosslistedCourses(json.duplicateCrosslistedCourses),
+      select: json => JSON.parse(json.configs[0].value),
+    },
+  );
+
+  const duplicateCourse = async () => {
+    const { course, crossListed, reserves } = data;
+    const { courseListingObject: courseListing } = course;
+
+    const newCourseListing = await ky.post('coursereserves/courselistings', { json: {
+      courseTypeId: courseListing.courseTypeId,
+      externalId: courseListing.externalId,
+      locationId: courseListing.locationId,
+      registrarId: courseListing.registrarId,
+      servicepointId: courseListing.servicepointId,
+      termId: term,
+    } }).json();
+
+    for (let i = 0; i < courseListing.instructorObjects.length; i++) {
```

Comment:
Doesn't this code wait for each add-instructor to be completed before beginning the next one? Maybe you need to do it this way to preserve the order, but otherwise I'd have expected to run them in parallel and collect the results with `Promise.all`.

Reply:
It does wait. I'm using this approach consciously (and even created a lint-rule PR to allow it) because of my negative experiences with machine-gun requests when writing batch importers. I actually just finished writing one to import our courses into Folio and chucking several requests at Folio at once wasn't stable enough for my taste - lots of ECONNRESET errors. Maybe I should put a comment in this code to that effect?

Reply:
Yes, a comment is a good idea — it would be all to easy for someone to "fix" this otherwise. But we should also take a moment to say: eek! If FOLIO can't handle half a dozen parallel requests, something is wrong.

Reply:
> Yes, a comment is a good idea — it would be all to easy for someone to "fix" this otherwise. But we should also take a moment to say: eek! If FOLIO can't handle half a dozen parallel requests, something is wrong.

I _do_ wonder whether it's more the client's connection is saturated by throwing hundreds of requests out at once in the importers scenario.

Reply:
If it were hundreds, I'd agree, but here we're just seeing two or three, aren't we?

(Also: I suppose a smart browser ought to do the right thing if it's given many nominally simultaneous requests, and throttle them through at a rate it knows it can handle. In fact, it's hard to believe browsers _don't_ already do this, since they must constantly run into this problem when loading HTML pages full of images, scripts and stylesheets.)

Reply:
Yeah...I wrote a big comment but now I'm gonna go back to refactor it and test it more. Maybe it is unnecessarily safe. 🔬🧪

Reply:
Ah, dude, I'm sorry, I didn't mean to make extra work for you.

Why don't you go ahead and merge this one, and do that separately?

Reply:
I think it's a problem if I find myself writing a comment addressed to Future Developer to do something that I'm too lazy to do. 😝 

I went ahead and tried it out against `snapshot`, looks like it works properly with a course that has 10 reserves, 10 instructors, and 10 crosslists so I pushed up a refactored version to this PR.
![Screen Shot 2021-03-02 at 10 30 13 AM](https://user-images.githubusercontent.com/5324374/109671671-47f18280-7b42-11eb-9141-067034e26616.png)

Reply:
> I think it's a problem if I find myself writing a comment addressed to Future Developer to do something that I'm too lazy to do. 😝

RIght, but initially at least you _chose_ not to do it for reasons other than laziness. I'l look at the new commit, but now I am worrying about what happens when we run into a course that that 20 instructors or 30 reserves of 40 cross-listings.

Reply:
BTW., why is that course now called "Way Too Many 101 - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate - Duplicate"?

Reply:
Because I duplicated a course that was duplicated from a course that was duplicated from a course...

Reply:
OK, good — the software didn't do it, _you_ did.